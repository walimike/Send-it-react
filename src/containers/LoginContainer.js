import React, { Component } from 'react';
import { connect } from 'react-redux';

import LoginModal from "../components/login/loginModal";
import loginAction from "../actions/loginAction";

import "../assets/styles/modal.scss";

 export class LoginContainer extends Component {
    constructor(props){
        super(props);
        this.state = {
            name:'',
            password:'',
            isLoading:false
        }
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps){
            this.setState({isLoading:false})
            if (nextProps.token) {
            sessionStorage.setItem('token', nextProps.token);
            setTimeout(() => window.location.replace('/home'), 100);
        }}
      }

    nameChange = (event) => this.setState({name: event.target.value});
    passwordChange = (event) => this.setState({password: event.target.value});

     handleSubmit = (event) => {
        event.preventDefault();
        this.setState({isLoading: true})
        this.logIn()
    }

    logIn = () => {
        const { loginAction } = this.props;
        const loginData = {
            name:this.state.name,
            password:this.state.password
        }
        loginAction(loginData);
    }

   render() {

     const {open} = this.props;
    const {close} = this.props;

     return (
      <LoginModal
      open = {open}
      close = {close}
      name={this.state.name}
      password={this.state.password}
      nameChange = {this.nameChange}
      passwordChange = {this.passwordChange}
      handleSubmit = {this.handleSubmit}
      isLoading = {this.state.isLoading}
       />
    )
  }
}


 LoginContainer.defaultProps = {
    token: '',
    errors: {},
  };

 const mapStateToProps = state => ({
    token: state.login.token,
    errors: state.login.errors,
    isSuccessful: state.login.isSuccessful,
  });

 export default connect(mapStateToProps, { loginAction })(LoginContainer);