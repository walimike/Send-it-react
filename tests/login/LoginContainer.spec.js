import configureStore from 'redux-mock-store';
import React from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import moxios from 'moxios';

import LoginContainer, {
  LoginContainer as App,
} from "../../src/containers/LoginContainer";

describe('LoginContainer', () => {
  const initialState = {
    login: { isSuccesfull: false, token: '', errors: null },
  };

  const mockFn = jest.fn();
  const mockStore = configureStore([thunk]);
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
    moxios.install();
  });

  afterEach(function() {
    moxios.uninstall();
  });

  it('should render without crashing', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <LoginContainer open={false} close={mockFn}  />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when state is updated', () => {
    const wrapper = mount(
      <Provider store={store}>
        <LoginContainer open={false} close={mockFn} />
      </Provider>
    );
    expect(wrapper.find('.modal').length).toEqual(1);
  });

  it('should call the login action', () => {
    const wrapper = shallow(<App open={false} close={mockFn} loginAction={mockFn} />);
    wrapper.setProps({ open: false, close: mockFn });

    const event = {
      preventDefault: mockFn,
      target: {
        elements: {
          name: {
            value: 'myname',
          },
          password: {
            value: 'Password1',
          },
        },
      },
    };
    wrapper.instance().handleSubmit(event);
  });

  it('pass next props',() => {
    const wrapper = shallow(<App open={false} close={mockFn} />);
    const nextProps = {token:'abcdefg'}
    wrapper.instance().componentWillReceiveProps(nextProps)
  });

  it('should call email',
    () => {
      const wrapper = shallow(<App open={false} close={mockFn} />);
      const nameEvent ={
          target:{value:''}
      }
      wrapper.instance().nameChange(nameEvent);
    });

    it('should call password',
    () => {
      const wrapper = shallow(<App open={false} close={mockFn} />);
      const passwordEvent ={
        target:{value:''}
    }
      wrapper.instance().passwordChange(passwordEvent);
    });
});