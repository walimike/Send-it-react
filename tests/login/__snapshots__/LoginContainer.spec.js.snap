// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer should render without crashing 1`] = `ShallowWrapper {}`;
